packages/app/src/vis-packs/core/utils.ts

```diff
@@ -112,6 +119,20 @@ export function toNumArray(
   return arr;
 }
 
+export function getPhaseAmp(
+  arr: ArrayValue<ComplexType>,
+): [phase: NumArray, amplitude: NumArray] {
+  const phase: number[] = Array.from({ length: arr.length });
+  const amplitude: number[] = Array.from({ length: arr.length });
+
+  arr.forEach(([real, imag], i) => {
+    phase[i] = Math.atan2(imag, real);
+    amplitude[i] = Math.hypot(real, imag);
+  });
```

Comment:
Trying to be a bit more efficient here by initialising the two arrays to the right length and assigning values to them instead of growing the arrays by pushing new values to them.